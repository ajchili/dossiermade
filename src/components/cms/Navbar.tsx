import React, { Component } from "react";
import { Link, withRouter, RouteComponentProps } from "react-router-dom";
import firebase from "../../lib/firebase";

interface Props extends RouteComponentProps {}

class Navbar extends Component<Props, {}> {
  _logout = () => {
    firebase.auth().signOut();
  };

  render() {
    return (
      <div uk-sticky="sel-target: .uk-navbar-container; cls-active: uk-navbar-sticky;">
        <nav className="uk-navbar-container" uk-navbar="true">
          <div className="uk-navbar-left">
            <Link to="/dashboard" className="uk-navbar-item uk-logo">
              DOSSIER MADE MEDIA - CMS
            </Link>
          </div>
          <div className="uk-navbar-right">
            <ul className="uk-navbar-nav">
              <li>
                <Link to="/">
                  Home
                </Link>
              </li>
              <li>
                <a href="#logout" onClick={this._logout}>
                  Logout
                </a>
              </li>
            </ul>
          </div>
        </nav>
      </div>
    );
  }
}

export default withRouter(Navbar);
