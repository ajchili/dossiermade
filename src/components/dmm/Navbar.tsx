import React, { Component } from "react";
import Menu from "./NavbarMenu";
import Medium from "./medium/Navbar";
import Small from "./small/Navbar";

export type Link = {
  name: string;
  url: string;
};

export interface Props {
  links: Array<Link>;
}

const links = [
  { name: "WORK", url: "work" },
  { name: "ABOUT", url: "about" },
  { name: "SERVICES", url: "services" },
  { name: "CONTACT", url: "contact" }
];

export default class extends Component {
  render() {
    return (
      <div uk-sticky="sel-target: .uk-navbar-container; cls-active: uk-navbar-sticky;">
        <div className="uk-hidden@s">
          <Small />
          <Menu links={links} />
        </div>
        <div className="uk-visible@s">
          <Medium links={links} />
        </div>
      </div>
    );
  }
}
