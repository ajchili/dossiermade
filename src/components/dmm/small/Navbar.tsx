import React, { Component } from "react";

export default class extends Component {
  render() {
    return (
      <nav className="uk-navbar-container uk-navbar-secondary uk-light">
        <div className="uk-navbar-left uk-background-secondary">
          <a
            className="uk-navbar-toggle"
            uk-toggle="target: #menu"
            href="#menu"
          >
            <span uk-navbar-toggle-icon="true" />
            <span className="uk-margin-small-left">DOSSIER MADE MEDIA</span>
          </a>
        </div>
      </nav>
    );
  }
}
