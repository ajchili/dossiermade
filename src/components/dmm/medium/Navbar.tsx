import React, { Component } from "react";
import { Link } from "react-router-dom";
import { Props } from "../Navbar";

export default class extends Component<Props> {
  render() {
    const { links } = this.props;

    return (
      <nav
        className="uk-navbar-container uk-navbar-secondary uk-light"
        uk-navbar="true"
      >
        <div className="uk-navbar-left uk-width-1-1 uk-background-secondary">
          <a href="#lander" className="uk-navbar-item uk-logo" uk-scroll="true">
            DOSSIER MADE MEDIA
          </a>
          <ul
            className="uk-navbar-nav"
            uk-scrollspy-nav="closest: li; scroll: true"
          >
            {links.map((link, i) => (
              <li key={i}>
                <a href={`#${link.url}`} uk-scroll="true">
                  {link.name}
                </a>
              </li>
            ))}
          </ul>
        </div>
        <div className="uk-navbar-right uk-background-secondary">
          <ul className="uk-navbar-nav">
            <li>
              <Link to="/login">Login</Link>
            </li>
          </ul>
        </div>
      </nav>
    );
  }
}
