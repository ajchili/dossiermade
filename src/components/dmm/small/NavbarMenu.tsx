import React, { Component } from "react";
import { Link } from "react-router-dom";
import { Props } from "../Navbar";

export default class extends Component<Props> {
  render() {
    const { links } = this.props;

    return (
      <div id="menu" uk-offcanvas="true">
        <div className="uk-offcanvas-bar">
          <button
            className="uk-offcanvas-close"
            type="button"
            uk-close="true"
          />
          <ul className="uk-nav">
            <li>
              <a
                className="uk-navbar-toggle"
                uk-toggle="target: #menu"
                href="#lander"
                uk-scroll="true"
              >
                HOME
              </a>
            </li>
            {links.map((link, i) => (
              <li key={i}>
                <a
                  className="uk-navbar-toggle"
                  uk-toggle="target: #menu"
                  href={`#${link.url}`}
                  uk-scroll="true"
                >
                  {link.name}
                </a>
              </li>
            ))}
            <li className="uk-hidden@s">
              <Link
                to="/login"
                className="uk-navbar-toggle"
              >
                LOGIN
              </Link>
            </li>
          </ul>
        </div>
      </div>
    );
  }
}
