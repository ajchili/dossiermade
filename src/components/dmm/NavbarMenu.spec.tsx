import * as React from "react";
import NavbarMenu from "./NavbarMenu";
import { Link } from "./Navbar";
import { shallow } from "enzyme";

describe("NavbarMenu", () => {
  it("renders no links if none are provided", () => {
    const tree = shallow(<NavbarMenu links={[]} />);
    expect(tree).toMatchSnapshot();
  });

  it("renders links if any are provided", () => {
    const links: Array<Link> = [
      {
        name: "TEST",
        url: "test"
      },
      {
        name: "TESTING",
        url: "testing"
      }
    ];
    const tree = shallow(<NavbarMenu links={links} />);
    expect(tree).toMatchSnapshot();
  });
});
