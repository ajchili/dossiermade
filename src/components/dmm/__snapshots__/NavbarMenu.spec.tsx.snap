// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarMenu renders links if any are provided 1`] = `
<div
  id="menu"
  uk-offcanvas="true"
>
  <div
    className="uk-offcanvas-bar"
  >
    <button
      className="uk-offcanvas-close"
      type="button"
      uk-close="true"
    />
    <ul
      className="uk-nav"
    >
      <li>
        <a
          className="uk-navbar-toggle"
          href="#lander"
          uk-scroll="true"
          uk-toggle="target: #menu"
        >
          HOME
        </a>
      </li>
      <li
        key="0"
      >
        <a
          className="uk-navbar-toggle"
          href="#test"
          uk-scroll="true"
          uk-toggle="target: #menu"
        >
          TEST
        </a>
      </li>
      <li
        key="1"
      >
        <a
          className="uk-navbar-toggle"
          href="#testing"
          uk-scroll="true"
          uk-toggle="target: #menu"
        >
          TESTING
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`NavbarMenu renders no links if none are provided 1`] = `
<div
  id="menu"
  uk-offcanvas="true"
>
  <div
    className="uk-offcanvas-bar"
  >
    <button
      className="uk-offcanvas-close"
      type="button"
      uk-close="true"
    />
    <ul
      className="uk-nav"
    >
      <li>
        <a
          className="uk-navbar-toggle"
          href="#lander"
          uk-scroll="true"
          uk-toggle="target: #menu"
        >
          HOME
        </a>
      </li>
    </ul>
  </div>
</div>
`;
